### ISSUE TITLE
Rayman 2 - black screen

### ISSUE THREAD

**Issue by deton24** [narzoul/DDrawCompat#30]
Game with system's ddraw.dll copied to game dir, on DX renderer runs only in windowed mode.
dgVoodoo decreases performance.
DDrawCompat causes black screen on startup.
W10
I saw runned game for a sec after alt tabbing.

_____
DXwrapper allows to run game smoothly in full screen after editing ini.
____

log

4048 00:53:20.117 Process path: C:\rayman 2\Rayman2.exe
4048 00:53:20.118 Environment variable __COMPAT_LAYER = "DWM8And16BitMitigation DetectorsXP"
4048 00:53:20.118 Loading DDrawCompat from H:\Gry\r2\DDRAW.dll
4048 00:53:20.119 Disabling external hook to DirectDrawCreate in apphelp.dll
4048 00:53:20.119 DDrawCompat loaded successfully
4048 00:53:20.167 Installing registry hooks
4048 00:53:20.167 Installing Direct3D driver hooks
4048 00:53:20.168 Installing DirectDraw hooks
4048 00:53:20.183 Hooking user mode display driver: "C:\Windows\System32\DriverStore\FileRepository\c0326037.inf_amd64_6cad8aeb5717c52d\B326079\aticfx32.dll"
4048 00:53:20.198 Installing Direct3D hooks
4048 00:53:20.202 Installing GDI hooks
4048 00:53:20.211 Installing display mode hooks
4048 00:53:20.211 Finished installing hooks
4048 00:53:20.213 Incorrect z-buffer bit depth capabilities detected for "AMD Radeon HD Series" / Direct3D HAL,
4048 00:53:20.213   changed supported z-buffer bit depths from "16, 32" to "16, 24"
8132 00:54:50.563 GDI used DC cache size: 1

**Comment by elishacloud** [comment]
@deton24, I made some updates to dxwrapper.  Try using this update here: [ddraw.zip](https://github.com/narzoul/DDrawCompat/files/3365387/ddraw.zip)

Just unzip the files into the Raymond 2 folder overwriting any existing file.

_Note: this was tested using the DirectX 6 Driver in the Raymond 2 configuration tool._

**Comment by deton24** [comment]
Is it something new to the update in previous thread we discussed?
https://github.com/elishacloud/dxwrapper/issues/35#issuecomment-500131861

What mysterious thing stays behind calling Rayman - Raymond? :)

**Comment by elishacloud** [comment]
This is a bit newer than that one, but essentially does the same thing.  Not sure why I keep calling it Raymond instead of Rayman...  LOL!

**Comment by deton24** [comment]
Thanks! I'll let you know when I perform some tests.

pon., 8 lip 2019, 01:00 użytkownik Elisha Riedlinger <
notifications@github.com> napisał:

**Comment by deton24** [comment]
I'm sorry but the game has very low performance now. But at least there is no broken graphics problem.

For now, this method works the best for me:
https://pcgamingwiki.com/wiki/Rayman_2:_The_Great_Escape#Game_runs_in_windowed_mode_.28fullscreen_fix.29
(ddraw.dll 1.0.2383.20, with its AllSettings, but dxwrapper.ini from beta 1 in linked thread)

And to be honest, the performance is not bad, but still not completely butter smooth like in nGlide (some FPS drops - e.g. like when I was saying about Fariy Glade, don't just go back in the bridge, but go to the very end up to the wall, and then turn back, look around).

I also checked your new official release, and it gives error of enumerating resolutions.
It happened even on linked here ddraw.dll (beta 2?), when I used AllSettings.ini different than in your beta 1 in linked thread.
Strangely, it even kept showing the same error once I replaced the ini.
Anyway, when use previously preserved copy of working R2+your ddraw.dll 1.0.2383.20, with its AllSettings and dxwrapper.ini beta 1 which fixes blurry graphics (like described in pcgamingwiki), it still works.
